### DISCUSSION TITLE
How to make custom wrapper for radio/checkbox to use inside group?

### DISCUSSION THREAD

**Discussion by 7iomka** [mantinedev/mantine#2134] [Q&A]
Hi.
I want to make and `RadioCard` component to work as `Radio` component, which is compatible to use both versions (controlled, uncontrolled) as child of Radio.Group

I try to make this one, but I cannot handle checked state.
I read the docs and examples for ui, and not found example to make this. Only example [with boolean values ](https://ui.mantine.dev/category/inputs#image-checkboxes), but it's not my case, because I use values.

Codesandbox: [https://codesandbox.io/s/fast-shape-q68zb8?file=/src/shared/ui/radio-card/radio-card.component.tsx](url)

**Comment by rtivital** [comment]
https://codesandbox.io/s/mystifying-meadow-4hj1cf?file=/src/App.tsx

**Reply by 7iomka** [reply]
> https://codesandbox.io/s/mystifying-meadow-4hj1cf?file=/src/App.tsx

this is a bad example
I want to keep standard flow how form's working with radio/checkbox group to put value and onChange to group.
Any other example?

**Reply by AabhasDhaubanja** [reply]
@rtivital is there a way to make it more accessible. How can I make the wrapper(UnstyledButton in this case) as the label for radio?

**Reply by rtivital** [reply]
No, Mantine does not provide such components, you need to build it on your side.

**Reply by AabhasDhaubanja** [reply]
Okay thanks

**Comment by 7iomka** [comment]
This example is not a wrapper around Radio/Checkbox because we don't make the same behavior.
Also, my question demonstrates the issue: we don't have access to Radio.Group provider in our custom component-wrappers if we want. `useRadioGroupContext` how I see is not public available

**Reply by rtivital** [reply]
Yes, Radio.Group can only work with Radio components. `useRadioGroupContext` and other similar hooks are not exposed as public api – I want to avoid creating a major release every time internal implementation changes. That said, I think you should build your own state for this case as the library does not provide anything for it out of the box.

**Comment by 7iomka** [comment]
@rtivital In other words, We cannot get radio/checkbox checked state in any component-wrapper, even if we can handle only variant of component (for example: uncontrolled) ?

**Reply by rtivital** [reply]
I do not understand the question. You can make Radio.Group controlled and get the state – https://mantine.dev/core/radio/#controlled-radiogroup

**Comment by 7iomka** [comment]
@rtivital Hi
Can I ask you to create a new component, like `radio-card` and/or `checkbox-card` that will use the radio/checkbox context inside and allow us 2 things
1) to hide the radio button itself visually (icon) (through styles api)
2) allow children prop to use instead of default `label` the custom content.

This kind of component will be very useful for creating decorative switches.
Currently, we do not have public access to the context to implement these components ourselves, and the implementation options proposed above are semantically not the right solution.


Example of usage cases
Simple:
<img width="426" alt="image" src="https://user-images.githubusercontent.com/18473137/185661718-ce661428-4fd0-41d5-ac61-43377bdf1ded.png">
<img width="418" alt="image" src="https://user-images.githubusercontent.com/18473137/185661776-7cc1e2f5-1727-44c7-8903-6822abdf7673.png">

With custom children
<img width="580" alt="image" src="https://user-images.githubusercontent.com/18473137/185661982-9ccdfeaf-c15c-455f-9a87-9a0fc8969007.png">

<img width="1184" alt="image" src="https://user-images.githubusercontent.com/18473137/185661916-44d4f3c8-191e-4ba7-afdc-67a8fed72c7a.png">

I would be very grateful to you.

**Reply by rtivital** [reply]
I'll consider that, but most likely it won't be added soon

**Reply by 7iomka** [reply]
> I'll consider that, but most likely it won't be added soon

maybe I could help in some way?
As far as I understand, you can take the implementation of the `SegmentedControl` as a basis and change the api a little.

**Reply by rtivital** [reply]
No, currently I do not have time to maintain this feature, I want to focus on other things first.
